temporal/server_impl.go

```diff
@@ -146,7 +146,13 @@ func (s *ServerImpl) Stop() error {
 }
 
 func (s *ServerImpl) startServices() error {
-	ctx, cancel := context.WithTimeout(context.Background(), serviceStartTimeout)
+	// The membership join time may exceed the configured max join duration.
+	// Double the service start timeout to make sure there is enough time for start logic.
+	serviceStartMaxTimeout := 2 * s.so.config.Global.Membership.MaxJoinDuration
+	if serviceStartMaxTimeout < serviceStartTimeout {
+		serviceStartMaxTimeout = serviceStartTimeout
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), serviceStartMaxTimeout)
```

Comment:
```suggestion
	timeout := util.Max(serviceStartTimeout, 2 * s.so.config.Global.Membership.MaxJoinDuration)
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```
?

(I know, I just made a min/max bug but this one is pretty clear)

Reply:
Kudos for this function.